File: typescipt/src/compiler/cSharp/writers/typeWriter.ts


export class TypeWriter extends IRootTokenWriter {
   public createCode(node: IRootNode): GeneratedClass {
     if (node is not TypeDefinition typeDefinition) throw new Error(`Root token not table`);

     let className = ClassNames.TypeClassName(typeDefinition.Name.Value);

     let classDeclaration = VariableClassFactory.TranslateVariablesClass(className, typeDefinition.Variables);

     return new GeneratedClass(node, className, classDeclaration);
   }
}

File: src/Lexy.Compiler/Compiler/CSharp/Writers/TypeWriter.cs
using System;
using Lexy.Compiler.Compiler.CSharp.Syntax;
using Lexy.Compiler.Language;
using Lexy.Compiler.Language.Tables;
using Lexy.Compiler.Language.Types;

namespace Lexy.Compiler.Compiler.CSharp.Writers;

internal class TypeWriter : IRootTokenWriter
{
    public GeneratedClass CreateCode(IRootNode node)
    {
        if (node is not TypeDefinition typeDefinition) throw new InvalidOperationException("Root token not type");

        var className = ClassNames.TypeClassName(typeDefinition.Name.Value);

        var classDeclaration = VariableClass.TranslateVariablesClass(className, typeDefinition.Variables);

        return new GeneratedClass(node, className, classDeclaration);
    }
}